src/workflow.ts

```diff
@@ -151,24 +151,29 @@ export class WorkflowContextImpl extends OperonContextImpl implements WorkflowCo
    */
   async flushResultBuffer(client: UserDatabaseClient): Promise<void> {
     const funcIDs = Array.from(this.resultBuffer.keys());
+    if (funcIDs.length === 0) {
+      return;
+    }
     funcIDs.sort();
     try {
+      let sqlStmt = "INSERT INTO operon.transaction_outputs (workflow_uuid, function_id, output, error, txn_id, txn_snapshot) VALUES ";
+      let paramCnt = 1;
+      const values: any[] = [];
       for (const funcID of funcIDs) {
```

Comment:
In my experience there is a limit to how big the batch should be... or it can take a lot of time / memory to make the string.